napalm_junos/junos.py

```diff
@@ -1325,3 +1325,26 @@ def get_optics(self):
             optics_detail[intf_optic_item[0]] = intf_optics
 
         return optics_detail
+
+    def get_config(self, retrieve='all'):
+        rv = {
+            'startup': '',
+            'running': '',
+            'candidate': ''
+        }
+
+        options = {
+            'format': 'text',
+            'database': 'candidate'
+        }
+
+        if retrieve in ('candidate', 'all'):
+            config = self.device.rpc.get_config(filter_xml=None, options=options)
```

Comment:
By definition the candidate config is only when in edit mode. It is equally true that the candidate config can be shared across multiple sessions, not limited to the current session only. So yes, I think your point makes sense to not restrict the gathering of the candidate config.